mitiq/pea/sample_amplifications.py

```diff
@@ -0,0 +1,126 @@
+from typing import Callable, List, Optional, Sequence, Union, cast
+
+import numpy as np
+from cirq import Circuit
+
+from mitiq import QPROGRAM, Executor, QuantumResult
+from mitiq.observable.observable import Observable
+from mitiq.pea.amplifications.amplify_depolarizing import (
+    amplify_noisy_ops_in_circuit_with_global_depolarizing_noise,
+    amplify_noisy_ops_in_circuit_with_local_depolarizing_noise,
+)
+from mitiq.pec import OperationRepresentation
+from mitiq.pec.sampling import sample_circuit
+
+
+def scale_circuit_amplifications(
+    ideal_circuit: Circuit,
+    scale_factors: float,
+    noise_model: str,
+    epsilon: float,
+) -> Sequence[OperationRepresentation]:
+    r"""Generates a list of implementable sequences from the noise-amplified
+    representation of the input ideal circuit based on the input noise model
+    and baseline noise level.
+
+    Args:
+        ideal_circuit: The ideal circuit from which an implementable
+            sequence is sampled.
+        scale_factors: A list of (positive) numbers by which the baseline
+            noise level is to be amplified.
+        noise_model: A string describing the noise model to be used for the
+            noise-scaled representations, e.g. "local_depolarizing" or
+            "global_depolarizing".
+        epsilon: Baseline noise level.
+
+    Returns:
+        A list of noise-amplified circuits, corresponding to each scale
+        factor multiplied by the baseline noise level."""
+
+    if noise_model == "local_depolarizing":
+        amp_fn = amplify_noisy_ops_in_circuit_with_local_depolarizing_noise
+        # TODO add other existing noise models from Mitiq
+    elif noise_model == "global_depolarizing":
+        amp_fn = amplify_noisy_ops_in_circuit_with_global_depolarizing_noise
+    else:
+        raise ValueError("Must specify supported noise model")
+        # TODO allow use of custom noise model
+
+    return amp_fn(ideal_circuit, (scale_factors - 1) * epsilon)
+
+
+def sample_circuit_amplifications(
+    ideal_circuit: Circuit,
+    executor: Union[Executor, Callable[[QPROGRAM], QuantumResult]],
+    scale_factors: List[float],
+    noise_model: str,
+    epsilon: float,
+    observable: Observable | None = None,
+    num_samples: Optional[int] = None,
+    force_run_all: bool = True,
+) -> List[float]:
```

Comment:
I'm a little confused by this function. It seems like it's orchestrating the everything except for the extrapolation.

Do we want to have this function given most of our techniques are available through either

1. `execute_with_xyz`, or
2. the two-step application of the technique with `construct_circuits` and `combine_results`?

I would think we would want to follow the same pattern here, and I'm not sure how this function fits into that pattern.